messenger.rst

```diff
@@ -124,46 +183,191 @@ sender. Part of a transport, it is responsible for sending your message somewher
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/messenger.yaml
+        framework:
+            messenger:
+                routing:
+                    'My\Message\ToBeSentToTwoSenders': [amqp, audit]
+
+    .. code-block:: xml
+
+        <!-- config/packages/messenger.xml -->
+        <container xmlns="http://symfony.com/schema/dic/symfony"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony
+                http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <framework:messenger>
+                    <framework:routing message-class="My\Message\ToBeSentToTwoSenders">
+                        <framework:sender service="amqp" />
+                        <framework:sender service="audit" />
+                    </framework:routing>
+                </framework:messenger>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/messenger.php
+        $container->loadFromExtension('framework', array(
+            'messenger' => array(
+                'routing' => array(
+                    'My\Message\ToBeSentToTwoSenders' => array('amqp', 'audit'),
+                ),
+            ),
+        ));
 
 By specifying a ``null`` sender, you can also route a class of messages to a sender
 while still having them passed to their respective handler:
 
-.. code-block:: yaml
-
-    framework:
-        messenger:
-            routing:
-                'My\Message\ThatIsGoingToBeSentAndHandledLocally': [amqp, ~]
-
-Consuming messages
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/messenger.yaml
+        framework:
+            messenger:
+                routing:
+                    'My\Message\ThatIsGoingToBeSentAndHandledLocally': [amqp, ~]
+
+    .. code-block:: xml
+
+        <!-- config/packages/messenger.xml -->
+        <container xmlns="http://symfony.com/schema/dic/symfony"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony
+                http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <framework:messenger>
+                    <framework:routing message-class="My\Message\ThatIsGoingToBeSentAndHandledLocally">
+                        <framework:sender service="amqp" />
+                        <framework:sender service="" />
```

Comment:
@javiereguiluz this one doesn't look right to me, but I fear we need to fix the code first to make the example possible